core/src/main/scala/iliad/SizeOf.scala

```diff
@@ -0,0 +1,13 @@
+package iliad
+
+import simulacrum.typeclass
+
+@typeclass
+trait SizeOf[A] {
```

Comment:
This should be different - I think it ought to contain the metadata to put stuff into a buffer. Possibly ByteInfo?